### ISSUE TITLE
Replicating CIFAR10 Accuracy

### ISSUE THREAD

**Issue by afzalxo** [uranusx86/BinaryNet-on-tensorflow#3]
Hi!I am trying to replicate the accuracy of CIFAR-10, stated to be 88.60% in the paper (Theano Implementation). You managed to achieve 86.18% which is still low compared to that stated in the paper. Do you have any ideas as to why we get reduced accuracy? And why did you use layer learning rates as layer_lr/4 instead of the baseline Glorot?
Thanks

**Comment by uranusx86** [comment]
I use layer learning rates as layer_lr/4 for simple because I don't know how to fetch the layer input size in the optimizer computing gradient stage.
And I have no idea what to do weight decay (note it's different from L2 regularization) in the Adam optimizer with tensorflow.
Maybe it's the point why we get reduced accuracy.

The BNN author have implemented binary net with tensorflow now, you can check it!
https://github.com/itayhubara/BinaryNet.tf

**Comment by afzalxo** [comment]
Hey! I think you're right. Non inclusion of layered weight decay could be the cause for the lowered accuracy. And thanks for pointing me to the repo with the TF implementation of the binarized convnet. I trained the cifar10 network from the linked repo (https://github.com/itayhubara/BinaryNet.tf) for about 250 epochs and this implementation does seem to reach about 88% accuracy.
I'll see if I can port it to my own implementation for my needs.
Thanks again! Issue resolved.
